dask_gdf/tests/test_accessor.py

```diff
@@ -5,46 +5,172 @@
-    gdf_data = Series(pd_data)
-    dask_gdf_data = dgd.from_pygdf(gdf_data, npartitions=5)
-    dask_gdf_data.dt
+    pdsr = pd.Series(data.copy())
+    sr = Series(pdsr)
+    dsr = dgd.from_pygdf(sr, npartitions=5)
+    dsr.dt
 
 
-@pytest.mark.parametrize('data', [data()])
+@pytest.mark.parametrize('data', [data_dt_1()])
 def test_series(data):
-    pd_data = pd.Series(data.copy())
-    gdf_data = Series(pd_data)
-    dask_gdf_data = dgd.from_pygdf(gdf_data, npartitions=5)
+    pdsr = pd.Series(data.copy())
+    sr = Series(pdsr)
+    dsr = dgd.from_pygdf(sr, npartitions=5)
 
     np.testing.assert_equal(
-        np.array(pd_data),
-        np.array(dask_gdf_data.compute()),
+        np.array(pdsr),
+        np.array(dsr.compute()),
     )
 
 
-@pytest.mark.parametrize('data', [data()])
-@pytest.mark.parametrize('field', fields)
+@pytest.mark.parametrize('data', [data_dt_1()])
+@pytest.mark.parametrize('field', dt_fields)
 def test_dt_series(data, field):
-    pd_data = pd.Series(data.copy())
-    gdf_data = Series(pd_data)
-    dask_gdf_data = dgd.from_pygdf(gdf_data, npartitions=5)
-    base = getattr(pd_data.dt, field)
-    test = getattr(dask_gdf_data.dt, field).compute()\
-                                           .to_pandas().astype('int64')
+    pdsr = pd.Series(data.copy())
+    sr = Series(pdsr)
+    dsr = dgd.from_pygdf(sr, npartitions=5)
+    base = getattr(pdsr.dt, field)
+    test = getattr(dsr.dt, field).compute()\
+                                 .to_pandas().astype('int64')
     assert_series_equal(base, test)
+
+
+#############################################################################
+#                        Categorical Accessor                               #
+#############################################################################
+
+
+def data_cat_1():
+    cat = pd.Categorical(['a', 'a', 'b', 'c', 'a'], categories=['a', 'b', 'c'])
+    return cat
+
+
+def data_cat_2():
+    return pd.Series([1, 2, 3])
+
+
+def data_cat_3():
+    cat1 = pd.Categorical(['a', 'a', 'b', 'c', 'a'],
+                          categories=['a', 'b', 'c'], ordered=True)
+    cat2 = pd.Categorical(['a', 'b', 'a', 'c', 'b'],
+                          categories=['a', 'b', 'c'], ordered=True)
+    return cat1, cat2
+
+
+@pytest.mark.parametrize('data', [data_cat_2()])
+@pytest.mark.xfail(raises=AttributeError)
+def test_categorical_accessor_initialization(data):
+    sr = Series(data.copy())
+    dsr = dgd.from_pygdf(sr, npartitions=5)
+    dsr.cat
+
+
+@pytest.mark.parametrize('data', [data_cat_1()])
+def test_categorical_basic(data):
+    cat = data.copy()
+    pdsr = pd.Series(cat)
+    sr = Series(cat)
+    dsr = dgd.from_pygdf(sr, npartitions=2)
+    result = dsr.compute()
+    np.testing.assert_array_equal(cat.codes, result.to_array())
+    assert dsr.dtype == pdsr.dtype
+
+    # Test attributes
+    assert pdsr.cat.ordered == dsr.cat.ordered.compute()
+    # TODO: Investigate dsr.cat.categories: It raises
```

Comment:
@kkraus14, It appears that the `categories` property is broken. I haven't had time to investigate the exact reason behind this. However, everything else works perfectly.